core/src/main/scala/fs2/job/JobManager.scala

```diff
@@ -228,8 +252,8 @@ object JobManager {
       : Stream[F, JobManager[F, I, N]] = {
 
     for {
-      notificationsQ <- Stream.eval(Queue.bounded[F, Option[N]](notificationsLimit))
-      eventQ <- Stream.eval(Queue.bounded[F, Option[Event[I]]](eventsLimit))
+      notificationsQ <- Stream.eval(Queue.bounded[F, Option[(I, N)]](notificationsLimit))
+      eventQ <- Stream.eval(Queue.circularBuffer[F, Option[Event[I]]](eventsLimit))
```

Comment:
`Event` use a `circularBuffer` so they never apply backpressure. Not sure if that's what we want.

Reply:
We could have them apply backpressure when something is consuming them and not otherwise.

Reply:
I'm honestly not sure if circular buffer is right here, but we're going to need to play around with it to see for sure. It means that we lose notifications when we otherwise would have slowed down. Only applying backpressure on consumption also loses data, and it introduces an initialization race condition where it can be difficult to reason about what you're going to get.

IMO we should try the circular buffer approach and see where it takes us. The most general approach is really to allow strategies to be chosen for various notifications, corresponding to a queue or a circular buffer. That's overkill for now though.